### ISSUE TITLE
Add a --proxy parameter to kotatsu serve

### ISSUE THREAD

**Issue by themouette** [Yomguithereal/kotatsu#78]
A full featured development experience requires both:

* `kotatsu start src/server.js`
* `kotatsu serve src/public.js`

Unless I missed something, a load balancer is required to expose both on a single port.

Adding a `--proxy` parameter would ease development workflow (for instance using [concurrently](https://www.npmjs.com/package/concurrently)):

``` sh
concurrent --kill-others \
  "kotatsu start src/server.js --port 1338" \
  "kotatsu serve src/public.js --port 1337 --proxy http://localhost:1338"
```

By the way I would suggest using [webpack-dev-server](https://webpack.github.io/docs/webpack-dev-server.html#api) instead of custom server, and forward `devServer` key from configuration in the webpack configuration.

**Comment by Yomguithereal** [comment]
Hello @themouette. Is this a problem to serve on two different ports? Also, you are using the `--port` option with the `start` command and this does nothing. The `start` command doesn't need a port. I guess the `server.js` scripts spawns an express app and you will set the port by using `app.listen(1338)`.

Concerning `webpack-dev-server` I use `webpack-dev-middleware` which has quite the same effect. I can use this middleware along with `webpack-hot-middleware` etc. to better customize the server offered by the tool if needed.

**Comment by Yomguithereal** [comment]
Apart from that, let me check if I can quickly add a `--proxy` option.

**Comment by themouette** [comment]
Hello @Yomguithereal,

thanks for the (very) quick answer. Indeed, `start` and `--port 1338` was a mistake, I tough it made the example easier to follow though.

Using a single port makes it more production like by providing a single entrypoint, but I get your point, CORS and configuration can do the trick.

I understand the flexibility of `webpack-dev-middleware` and `webpack-hot-middleware` and that's fine. The suggestion was based on the following thoughts:

* `webpack-dev-middleware` never access webpack config, so it makes it impossible to provide a custom configuration for it outside of cli options;
* `webpack-dev-server` seems to provide standard features;
* `webpack-dev-server` is widely used by the community.

**Comment by Yomguithereal** [comment]
Yes I tend to use CORS generally so I unintentionally discarded other options, which is bad. So I'll try to add a `--proxy` option.

I thought that the webpack team deprecated the `webpack-dev-server`. I may be wrong. Let me check whether it return an express app before switching to it (plus you are right it has a proxy option which could come in handy in your case).

**Comment by Yomguithereal** [comment]
I am reading the code of `webpack-dev-server` and I am not very fond of what it offers more than the middleware. However, I see that (apart from the proxy option, but I will fix that :) ), the middleware takes the same options as the server. So what I will do is make sure the `devServer` options of the config are passed to the middleware.

**Comment by themouette** [comment]
> So what I will do is make sure the devServer options of the config are passed to the middleware.

Looks good to me. thanks a lot.

**Comment by Yomguithereal** [comment]
@themouette, here is what I propose:

```bash
--proxy [namespace] [target]
# example
--proxy / http://localhost:8000
--proxy /api http://localhost:8000
```

**Comment by themouette** [comment]
You rock !

**Comment by Yomguithereal** [comment]
Just published `0.8.0` on npm. Tell me if this suits your needs :smile:.

**Comment by themouette** [comment]
that's perfect.
